test-data/unit/check-kwargs.test

```diff
@@ -500,3 +500,57 @@ def g(x=1): pass
 f(**{})
 g(**{})
 [builtins fixtures/dict.pyi]
+
+[case testKeywordUnpackWithDifferentTypes]
+# https://github.com/python/mypy/issues/11144
+from typing import Dict, Generic, TypeVar, Mapping
+
+T = TypeVar("T")
+T2 = TypeVar("T2")
+
+class A(Dict[T, T2]):
+    ...
+
+class B(Mapping[T, T2]):
+    ...
+
+class C(Generic[T, T2]):
+    ...
+
+class D:
+    ...
+
+def foo(**i: float) -> float:
+    ...
+
+a: A[str, str]
+b: B[str, str]
+c: C[str, float]
+d: D
+e = {"a": "b"}
+
+foo(k=1.5)
+foo(**a)
+foo(**b)
+foo(**c)
+foo(**d)
+foo(**e)
+
+# Correct:
+
+class Good(Mapping[str, float]):
+    ...
+
+good1: Good
+good2: A[str, float]
+good3: B[str, float]
+foo(**good1)
+foo(**good2)
+foo(**good3)
+[out]
+main:29: error: Argument 1 to "foo" has incompatible type "**A[str, str]"; expected "float"
```

Comment:
Can you update tests to use `# E: <message:` comments for error message?

Reply:
Sure! Will do 👍